x/cron/keeper/keeper.go

```diff
@@ -67,19 +67,27 @@ func (k *Keeper) Logger(ctx sdk.Context) log.Logger {
 // ExecuteReadySchedules gets all schedules that are due for execution (with limit that is equal to Params.Limit)
 // and executes messages in each one
 // NOTE that errors in contract calls rollback all already executed messages
-func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context) {
+func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context, executionStage types.ExecutionStage) {
 	telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), LabelExecuteReadySchedules)
 	schedules := k.getSchedulesReadyForExecution(ctx)
 
 	for _, schedule := range schedules {
-		err := k.executeSchedule(ctx, schedule)
-		recordExecutedSchedule(err, schedule)
+		if isExecutableAtTheStage(schedule, executionStage) {
+			err := k.executeSchedule(ctx, schedule)
+			recordExecutedSchedule(err, schedule)
+		}
 	}
 }
 
 // AddSchedule adds new schedule to execution for every block `period`.
```

Comment:
> for every block `period`.

that's also confusing. could you please rephrase it similarly to the other rephrasing I've asked you about? :)

Reply:
I've tried to rephrase

Reply:
Done